jtrust-lib/src/main/java/be/fedict/trust/linker/CustomCertSignValidator.java

```diff
@@ -0,0 +1,93 @@
+package be.fedict.trust.linker;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.security.*;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+/**
+ * Custom validator to validate the signature on a certificate.
+ */
+public class CustomCertSignValidator {
+
+    private static final Log LOG = LogFactory.getLog(CustomCertSignValidator.class);
+
+    /**
+     * Verifies that the public key of the given self-signed Root CA is used to sign itself.
+     *
+     * Try to verify the signature with the standard algorithm.
+     * If that doesn't succeed, the signature is verified with PKCS1 padding.
+     *
+     * @param certificate
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeyException
+     * @throws SignatureException
+     * @throws CertificateException
+     * @throws NoSuchProviderException
+     */
+    public static void verify(X509Certificate certificate) throws NoSuchAlgorithmException, InvalidKeyException, SignatureException, CertificateException, NoSuchProviderException {
+        verify(certificate, certificate);
+    }
+
+    /**
+     * Verifies that the public key in the first certificate is used to encrypt the second certificate.
```

Comment:
Verifies that the public key in the <code>certificate</code> is used to sign the <code>childCertificate</code>.

Reply:
Heb de beschrijving aangepast analoog naar de docs van java.security.cert.Certificate#verify.